### DISCUSSION TITLE
Does the order of operations matter?

### DISCUSSION THREAD

**Discussion by ChristopherZhong** [prisma/prisma#13170] [Q&A]
Using the user and posts examples from the official documentation, I have a `PUT` endpoint to update the users (e.g., `PUT /users/:id`). This method also allows the `posts` to be replaced. The code to do the `PUT` is as follows.

```typescript
prisma.user.update({
    where: { id: id },
    include: { posts: true },
    data: {
        // other fields removed for brevity
        posts: {
            deleteMany: [{ userId: id }],
            create: [...],
        },
    },
});
```

However, I noticed that if I swap the `deleteMany` and `create`, an exception occurs (``Unique constraint failed on the fields: (`userId`,`postId`)``).

```typescript
        posts: {
            create: [...],
            deleteMany: [{ userId: id }],
        },
```

Based on the exception, it appears that order does matter. However, I was unable to locate where in the documentation it is mentioned. If the order matters, does it also means that it matter everywhere? For instance.

```typescript
prisma.user.update({
    data: { ... },
    include: { ... },
    where: { ... },
});
```

**Comment by pantharshit00** [comment]
Hello @ChristopherZhong , our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

The order should not matter here. If you can still reproduce this, please open an issue here: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml

**Reply by vladi-stevanovic** [reply]
👋 Hello @ChristopherZhong I'm going through pending GitHub Discussions to check if you need any help. Have you perhaps had a chance to open a GitHub issue?

Please let share the link when you create it, so that we can continue to investigate this issue! 💯

**Reply by ChristopherZhong** [reply]
It has been a while and I no longer have the means to attempt to reproduce this quickly

**Reply by vladi-stevanovic** [reply]
Thank you for following up and letting us know. I appreciate you can't currently reproduce it, but if you come across the same or similar issue in the future, don't hesitate to let us know!

**Reply by i-Portillo** [reply]
Hi 👋 I've recently run into this issue and managed to quickly fire up a small project that showcases the problem: https://github.com/i-Portillo/prisma-bug-repro

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
